### DISCUSSION TITLE
How can I ensure my gt table sits on its own page (when outputting as pdf)?

### DISCUSSION THREAD

**Discussion by e05bf027** [rstudio/gt#1467] [Q&A]
I am writing a thesis. There is a table that I have generated using `gtsummary`. My output is a pdf through Quarto markdown.

At the end of the workflow I want to make it output `as_latex()`. The table itself looks perfect, but I want to avoid it splitting across different pages (the table itself is not too long for a single page). Essentially I want to create a floating environment in the same way as a LaTeX document, but I am unsure how to do that in this context.

Code for  my table:

```
demo_summary%>% 
  select(-c(patient_id,
            pathogenic_factor,
            ards_risk_factor,
            adm_date,
            # adm_location,
            dc_date_time,
            ventilator)) %>% 
  relocate(bmi, .after = weight_kg) %>% 
  tbl_summary(by = NULL,
              statistic = list(all_continuous() ~ "{mean} ({sd})",
                               all_categorical() ~ "{n} ({p}%)"),
              digits = list(all_continuous() ~ 1,
                            all_categorical() ~ 0),
              label = list(gender ~ 'Gender',
                           age_years ~ 'Age (years)',
                           height_cm ~ 'Height (cm)',
                           weight_kg ~ 'Weight (kg)',
                           bmi ~ 'BMI',
                           ards_type ~ 'ARDS classification',
                           adm_location ~ 'Admitting location',
                           los_days ~ 'LOS (days)',
                           apache_ii ~ 'Apache II',
                           total_proning_sessions ~ 'Proning sessions',
                           outcome ~ 'Outcome'),
              missing = 'ifany',
              missing_text = 'Not recorded') %>%
  bold_labels() %>% 
  # add_overall() %>% 
  as_gt() %>% 
  as_latex()
```

I asked about this from the point of view of Quarto but they said it wasn't something that could be facilitated from their point of view, but that I should try here directly.

Any idea how I would do this?

**Comment by benyamindsmith** [comment]
I am interested in this question as well.

**Comment by reddylee** [comment]
Have you fixed it out? I am curious about it as well.

**Reply by e05bf027** [reply]
Unfortunately not!

**Reply by hlageek** [reply]
I think adding LaTeX `\newpage` command into quarto document could do the trick:

Lore ipsum.
\newpage

````
```{r}
#| label: table_label
#| echo: false
#| output: asis

demo_summary%>% 
  select(-c(patient_id,
            pathogenic_factor,
            ards_risk_factor,
            adm_date,
            # adm_location,
            dc_date_time,
            ventilator)) %>% 
  relocate(bmi, .after = weight_kg) %>% 
  tbl_summary(by = NULL,
              statistic = list(all_continuous() ~ "{mean} ({sd})",
                               all_categorical() ~ "{n} ({p}%)"),
              digits = list(all_continuous() ~ 1,
                            all_categorical() ~ 0),
              label = list(gender ~ 'Gender',
                           age_years ~ 'Age (years)',
                           height_cm ~ 'Height (cm)',
                           weight_kg ~ 'Weight (kg)',
                           bmi ~ 'BMI',
                           ards_type ~ 'ARDS classification',
                           adm_location ~ 'Admitting location',
                           los_days ~ 'LOS (days)',
                           apache_ii ~ 'Apache II',
                           total_proning_sessions ~ 'Proning sessions',
                           outcome ~ 'Outcome'),
              missing = 'ifany',
              missing_text = 'Not recorded') %>%
  bold_labels() %>% 
  # add_overall() %>% 
  as_gt() %>% 
  as_latex()

```
````

\newpage

Lore ipsum.

**Reply by reddylee** [reply]
It works to some extent but isn't perfect. After the `\newpage` command, it may leave a lengthy blank page. By using floating, the table will automatically move to the next page, allowing the text below it to shift up, preventing the creation of a large blank space.

**Reply by hlageek** [reply]
It is possible to exercise more control over the LaTeX output by hijacking the environment with something like this: `cat(gsub("longtable", "tabular", gt::as_latex(gt_table)))` at the codechunk end. I have been using this e.g. to smuggle page-wide tables into two-column format. But it feels and acts very hacky.

**Reply by reddylee** [reply]
Thanks, it works to move the table to the next page, but the text below it does not shift up  to fill the blank in the previous page.

**Reply by hlageek** [reply]
Try adding `\begin{table}` before and `\end{table}` after the codechunk. Also you may need to add the package to quarto header:
```
format:
  pdf:
    include-in-header:
      text: |
        \usepackage{tabular}
```
Basically, after hacking the `gt::as_latex()` output, the whole question becomes a pure LaTeX question of how to format documents.

**Reply by reddylee** [reply]
I am not good at latex and failed in many attempts. But thanks anyway.

**Reply by olivroy** [reply]
Hi both, please give a try to the development version of gt. gt tables are now rendered as tabular instead of longtable

```
pak::pak("rstudio/gt")
```

see changelog for available customization

**Reply by reddylee** [reply]
Thanks, Olivroy. It works now. Excellent!

**Reply by reddylee** [reply]
I got another issue: it looks like the following function does not work in pdf. I want to increase the height of rows but failed.

`tab_options(data_row.padding = px(35), 
              heading.padding = px(35)) `

**Reply by olivroy** [reply]
@reddylee Please open an issue about this! Follow the template and provide a reproducible example!

**Reply by reddylee** [reply]
Thanks.
